receiver/prometheusreceiver/internal/metricsbuilder_test.go

```diff
@@ -1205,6 +1205,15 @@ func Test_isUsefulLabel(t *testing.T) {
 	}
 }
 
+func Benchmark_dpgSignature(b *testing.B) {
+	knownLabelKeys := []string{"a", "b"}
+	labels := labels.FromStrings("a", "va", "b", "vb", "x", "xa")
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		dpgSignature(knownLabelKeys, labels)
```

Comment:
I think the compiler may be smarter and avoid this call, so we need to save the result and say just compare the len to what we expect to be.

Reply:
I don't think it was happening when I ran this benchmark, but added a call to runtime.KeepAlive(), which seems to be current best practice for dealing with this issue per https://golang.org/issue/27400. Benchmark result is pretty much the same after this change.

Reply:
That's why I said "may be", was not sure :)

Reply:
Right :) Anyway, keeping KeepAlive call here in case Golang compiler will get more aggressive when optmizing in the future.